scripts/docs.sh

```diff
@@ -29,6 +29,6 @@
 set -e
 scripts/update_robotstxt.sh
 cd docs
-pip3 install -r requirements.txt
+pip3 install -r requirements.txt --upgrade --upgrade-strategy eager
```

Comment:
This is the whole point of the PR. Default upgrade strategy is `only-if-needed` so if an older package happens to be already installed and fits the requirements but a newer version is also available, `pip` keeps the older one.

Looks like we install the outdated `python3-sphinx` ourselves in the [Ubuntu image](https://github.com/ethereum/solidity/blob/v0.8.6/scripts/docker/buildpack-deps/Dockerfile.ubuntu2004#L42). I think we can safely remove it from there.